Dockerfile

```diff
@@ -1,60 +1,54 @@
-FROM openjdk:11-jdk
+FROM maven:3.8-jdk-11-slim as build
 
 ENV MAVEN_OPTS="-Dmaven.repo.local=.m2/repository -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=WARN -Dorg.slf4j.simpleLogger.showDateTime=true -Djava.awt.headless=true"
 ENV MAVEN_CLI_OPTS="--batch-mode --errors --fail-at-end --show-version -DinstallAtEnd=true -DdeployAtEnd=true"
 
 ARG ORS_CONFIG=./openrouteservice/src/main/resources/ors-config-sample.json
 ARG OSM_FILE=./openrouteservice/src/main/files/heidelberg.osm.gz
 ENV BUILD_GRAPHS="False"
-ARG UID=1000
 ARG TOMCAT_VERSION=8.5.69
 
-# Create user
-RUN useradd -u $UID -md /ors-core ors
-
-# Create directories
-RUN mkdir -p /usr/local/tomcat /ors-conf /var/log/ors && \
-    chown ors:ors /usr/local/tomcat /ors-conf /var/log/ors
+WORKDIR /ors-core
 
 # Install dependencies and locales
 RUN apt-get update -qq && \
-    apt-get install -qq -y locales nano maven moreutils jq && \
-    rm -rf /var/lib/apt/lists/* && \
-    locale-gen en_US.UTF-8
-
-USER ors:ors
-WORKDIR /ors-core
+    apt-get install -qq -y nano moreutils jq wget
 
-COPY --chown=ors:ors openrouteservice /ors-core/openrouteservice
-COPY --chown=ors:ors $OSM_FILE /ors-core/data/osm_file.pbf
-COPY --chown=ors:ors $ORS_CONFIG /ors-core/openrouteservice/src/main/resources/ors-config-sample.json
-COPY --chown=ors:ors ./docker-entrypoint.sh /ors-core/docker-entrypoint.sh
+COPY openrouteservice /ors-core/openrouteservice
+COPY $OSM_FILE /ors-core/data/osm_file.pbf
+COPY $ORS_CONFIG /ors-core/openrouteservice/src/main/resources/ors-config-sample.json
 
 # Install tomcat
 RUN wget -q https://archive.apache.org/dist/tomcat/tomcat-8/v${TOMCAT_VERSION}/bin/apache-tomcat-${TOMCAT_VERSION}.tar.gz -O /tmp/tomcat.tar.gz && \
     cd /tmp && \
     tar xvfz tomcat.tar.gz && \
+    mkdir /usr/local/tomcat && \
     cp -R /tmp/apache-tomcat-${TOMCAT_VERSION}/* /usr/local/tomcat/ && \
     rm -r /tmp/tomcat.tar.gz /tmp/apache-tomcat-${TOMCAT_VERSION}
 
-# Configure ors config
+# Configure ors config:
+# - Replace paths in ors-config.json to match docker setup
+# - init_threads = 1, > 1 been reported some issues
+# - Delete all profiles but car
 RUN cp /ors-core/openrouteservice/src/main/resources/ors-config-sample.json /ors-core/openrouteservice/src/main/resources/ors-config.json && \
-    # Replace paths in ors-config.json to match docker setup
```

Comment:
docker interprets them as empty lines and raises a warning in newer versions.

Reply:
Nice catch. Besides, it could be worth adding `SHELL ["/bin/bash", "-o", "pipefail", "-c"]` before the run, since we have plenty of pipes in it. Else errors during the piping are never caught, except the last exit code. For reference:

https://github-wiki-see.page/m/hadolint/hadolint/wiki/DL4006